client/src/components/Listings/ProfileCard/ProfileCard.tsx

```diff
@@ -30,6 +30,7 @@ export default function ProfileCard({ user }: Props): JSX.Element {
       }
     });
   };
+  if (!profile) return <></>;
```

Comment:
Instead of having this logic here, I would render the logic in its parent component like `profile && <Profile Card />` since its just returning an empty element here